packages/medusa/src/services/event-bus.ts

```diff
@@ -90,25 +80,35 @@ export default class EventBusService {
 
     if (singleton) {
       const opts = {
-        createClient: (type: string): Redis.Redis => {
+        createClient: (type: string, redisOpts: RedisOptions) => {
           switch (type) {
             case "client":
               return redisClient
             case "subscriber":
               return redisSubscriber
             default:
               if (config.projectConfig.redis_url) {
-                return new Redis(config.projectConfig.redis_url)
+                return new Redis(config.projectConfig.redis_url, {
+                  ...(redisOpts ?? {}),
+                  // Required settings. See: https://github.com/OptimalBits/bull/issues/1873
+                  maxRetriesPerRequest: null,
+                  enableReadyCheck: false,
+                })
               }
               return redisClient
           }
         },
       }
 
       this.eventToSubscribersMap_ = new Map()
-      this.queue_ = new Bull(`${this.constructor.name}:queue`, opts)
+      this.queue_ = new Queue(`queue`, {
+        prefix: `${this.constructor.name}`,
+        ...(opts as any),
```

Comment:
todo: we should cast with the expected type instead of any right? is there any typing issues when we don't cast?